src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1257,6 +1273,56 @@ class Config:
         datastore = 'vm.device'
         datastore_extend = 'vm.device.extend_device'
 
+    @private
+    async def nic_check(self, vm_device):
+        if not await self.middleware.call('system.is_freenas') and await self.middleware.call('failover.licensed'):
+            failover_enabled = not (await self.middleware.call('failover.config'))['disabled']
+            nics = await self.nic_capability_checks([vm_device])
+            if nics:
+                if failover_enabled:
+                    raise CallError(
+                        'Failover must be disabled before attempting to create/update VM NIC device '
+                        f'because of the following nics: {",".join(nics)}', errno.EPERM
+                    )
+                else:
+                    for nic in nics:
+                        await self.middleware.call('interface.disable_capabilities', nic, nics[nic])
+                        try:
+                            await self.middleware.call(
+                                'failover.call_remote', 'interface.disable_capabilities', [nic, nics[nic]]
+                            )
+                        except Exception as e:
+                            self.middleware.logger.debug(
+                                f'Failed to disable capabilities for {nic} on remote node: {e}'
+                            )
+
+    @private
+    async def nic_capability_checks(self, vm_devices=None, check_system_iface=True):
+        """
+        For NIC devices, if VM is started and NIC is added to a bridge, if desired nic_attach NIC has certain
+        capabilities set, we experience a hiccup in the network traffic which for failover can result in backup node
```

Comment:
To be consistent with the docstring in the jail plugin, let's change it to `we experience a hiccup in the network traffic which can cause a failover to occur.`